### PR TITLE
Setup publishing to LI artifactory

### PR DESCRIPTION
**Pull request by jplaisted**
Bintray is going to be deprecated soon, prepare to switch to our artifactory instance.

## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: docs/how/releases/continuous-releases.md [hunk 1]
```diff
@@ -29,8 +29,8 @@ We use shipkit changelog to create the release notes of each version. See
 ## Automatic Publishing
 
 Automation is done with a [GitHub workflow](../../.github/workflows/gh-version.yml) that will create the GitHub release
-and then publish to bintray for every push of the `main` branch.
+and then publish to artifactory for every push of the `main` branch.
```
Comment (camelliazhang) [inline docs/how/releases/continuous-releases.md:32]: Seems that we will stay in dual publishing mode for a while ? I see both bintrayUploadAll and  artifactoryPublishAll. Does it make sense to keep both mentioned here as well?

Comment (jplaisted) [reply docs/how/releases/continuous-releases.md:32]: So far as I'm concerned "awhile" can be just this PR's version lol. I don't think we need to dual publish for very long, so its not worth having the documentation reflect that, imo.

Comment (jplaisted) [reply docs/how/releases/continuous-releases.md:32]: Even if we do want to bake it awhile, I'm fine pointing new people just to the new area.

Comment (camelliazhang) [reply docs/how/releases/continuous-releases.md:32]: Prefer consistency, but agree it's fine for this particular case.

**Review by camelliazhang — APPROVED** [review]

**Comment by jplaisted** [comment]
https://github.com/linkedin/datahub-gma/issues/77

**Pull request merged**